$OpenBSD: patch-src_main_c,v 1.25 2013/05/24 13:34:11 dcoppa Exp $

Make --volume work with and without softvol enabled
(upstream svn revision r2426)

Unbreak cover art support

Set pref_volume to stored volume when asound is disabled
(upstream svn revision r2427)

Fix problem with remembered softvol not being loaded
(upstream svn revision r2424)

--- src/main.c.orig	Mon Feb 18 15:58:47 2013
+++ src/main.c	Fri May 24 14:29:58 2013
@@ -65,7 +65,6 @@ static gchar *tv_input;
 static gint tv_width;
 static gint tv_height;
 static gint tv_fps;
-static gint pref_volume;
 
 typedef struct _PlayData {
     gchar uri[4096];
@@ -344,7 +343,6 @@ gint play_iter(GtkTreeIter * playiter, gint restart_se
     while (gtk_events_pending())
         gtk_main_iteration();
 
-    /*
        // wait for metadata to be available on this item
        if (!streaming_media(uri) && !device_name(uri)) {
        i = 0;
@@ -364,7 +362,7 @@ gint play_iter(GtkTreeIter * playiter, gint restart_se
        &artist, ALBUM_COLUMN, &album, AUDIO_CODEC_COLUMN,
        &audio_codec, VIDEO_CODEC_COLUMN, &video_codec,
        VIDEO_WIDTH_COLUMN, &width, VIDEO_HEIGHT_COLUMN, &height,
-       DEMUXER_COLUMN, &demuxer, COVERART_COLUMN, &pixbuf,
+       DEMUXER_COLUMN, &demuxer, COVERART_COLUMN, &cover_art_file,
        SUBTITLE_COLUMN, &subtitle, AUDIOFILE_COLUMN, &audiofile,
        COUNT_COLUMN, &count, PLAYLIST_COLUMN, &playlist,
        PLAYABLE_COLUMN, &playable, -1);
@@ -392,7 +390,6 @@ gint play_iter(GtkTreeIter * playiter, gint restart_se
        }
 
        }
-     */
     // reset audio meter
     for (i = 0; i < METER_BARS; i++) {
         buckets[i] = 0;
@@ -900,6 +897,9 @@ int main(int argc, char *argv[])
     pplevel = gm_pref_store_get_int(gm_store, PPLEVEL);
 #ifndef HAVE_ASOUNDLIB
     volume = gm_pref_store_get_int(gm_store, VOLUME);
+    if (pref_volume == -1) {
+        pref_volume = volume;
+    }
 #endif
     audio_channels = gm_pref_store_get_int(gm_store, AUDIO_CHANNELS);
     use_hw_audio = gm_pref_store_get_boolean(gm_store, USE_HW_AUDIO);
@@ -1042,7 +1042,9 @@ int main(int argc, char *argv[])
         printf(_("gmtk v%s\n"), gmtk_version());
         printf("GTK %i.%i.%i\n", GTK_MAJOR_VERSION, GTK_MINOR_VERSION, GTK_MICRO_VERSION);
         printf("GLIB %i.%i.%i\n", GLIB_MAJOR_VERSION, GLIB_MINOR_VERSION, GLIB_MICRO_VERSION);
+#ifdef LIBGDA_ENABLED
         printf("GDA Enabled\n");
+#endif
     }
 
     if (cache_size == 0)
@@ -1084,6 +1086,7 @@ int main(int argc, char *argv[])
         if (remember_softvol && volume_softvol != -1) {
             gm_log(verbose, G_LOG_LEVEL_INFO, "Using last volume of %f%%", volume_softvol * 100.0);
             volume = (gdouble) volume_softvol *100.0;
+            audio_device.volume = volume / 100.0;
         } else {
             volume = 100.0;
         }
@@ -1308,16 +1311,18 @@ int main(int argc, char *argv[])
     // disabling this line seems to help with hangs on startup when using pulseaudio
     //gm_audio_get_volume(&audio_device);
     set_media_player_attributes(media);
-    if (!softvol) {
+    if (softvol) {
         if (pref_volume != -1) {
             audio_device.volume = (gdouble) pref_volume / 100.0;
             gm_log(verbose, G_LOG_LEVEL_INFO, "The volume on '%s' is %f", audio_device.description,
                    audio_device.volume);
             volume = audio_device.volume * 100;
+        } else {
+            audio_device.volume = volume / 100.0;
         }
-    } else {
-        audio_device.volume = volume / 100.0;
     }
+    gm_log(verbose, G_LOG_LEVEL_DEBUG, "Volume is %lf Audio Device Volume = %f", volume, audio_device.volume);
+
 #ifdef GTK2_12_ENABLED
     gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider), audio_device.volume);
 #else
